akka-stream-tests/src/test/scala/akka/stream/io/TlsSpec.scala

```diff
@@ -364,7 +389,8 @@ class TlsSpec extends StreamSpec("akka.loglevel=INFO\nakka.actor.debug.receive=o
             .scan(ByteString.empty)(_ ++ _)
             .via(new Timeout(6.seconds))
             .dropWhile(_.size < scenario.output.size)
-            .runWith(Sink.head)
+            .map(bs => {ks.shutdown(); bs})
```

Comment:
I think separate tests are better. We have had such leaks go unnoticed in other places (it's the reason for the `assertAllStagesStopped` wrapper around the test case.